src/main/java/org/broadinstitute/consent/http/models/dataset_registration_v1/builder/ConsentGroupFromDataset.java

```diff
@@ -90,7 +90,7 @@ private String findStringDSPropValue(Set<DatasetProperty> props, String propName
     if (Objects.nonNull(props) && !props.isEmpty()) {
       return props
           .stream()
-          .filter(p -> p.getSchemaProperty().equalsIgnoreCase(propName))
+          .filter(p -> p.getSchemaProperty() != null && p.getSchemaProperty().equalsIgnoreCase(propName))
```

Comment:
This seems like a good fit for Apache StringUtils (ditto throughout).

```suggestion
          .filter(p -> StringUtils.equalsIgnoreCase(p.getSchemaProperty(), propName))
```